state/machine.go

```diff
@@ -1671,14 +1671,14 @@ func (m *Machine) setPrivateAddressOps(providerAddresses []address, machineAddre
 
 // SetProviderAddresses records any addresses related to the machine, sourced
 // by asking the provider.
-func (m *Machine) SetProviderAddresses(addresses ...corenetwork.Address) error {
+func (m *Machine) SetProviderAddresses(addresses ...corenetwork.SpaceAddress) error {
 	err := m.setAddresses(nil, &addresses)
 	return errors.Annotatef(err, "cannot set addresses of machine %v", m)
 }
 
 // ProviderAddresses returns any hostnames and ips associated with a machine,
 // as determined by asking the provider.
-func (m *Machine) ProviderAddresses() (addresses []corenetwork.Address) {
+func (m *Machine) ProviderAddresses() (addresses corenetwork.SpaceAddresses) {
```

Comment:
Because server-side they are stored reasoned about in `SpaceAddress` form (with space ID), such as for controller config space filtering.

They are translated if passed out to clients.